From 8d12420d99afd278da687a6a36be92147ab2d4f2 Mon Sep 17 00:00:00 2001
From: Paul Lemire <paul.lemire@kdab.com>
Date: Thu, 29 Jul 2021 08:27:21 +0200
Subject: [PATCH 27/33] CameraLens: use ParametersDirty rather than AllDirty

AllDirty would force a lot of recomputations that aren't needed.
When the CameraLens changes the only things we need to do are:
- update projection matrix in the shader and the frustum culling

Those things are performed in the RenderCommand update stage which
is triggered with ParametersDirty

Change-Id: I27241f3ec323182b19fca7e5528d851680eeec8c
Reviewed-by: Mike Krus <mike.krus@kdab.com>
(cherry picked from commit ca43cd97277132341676d10a515419a5390c9292)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/render/backend/cameralens.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/render/backend/cameralens.cpp b/src/render/backend/cameralens.cpp
index 9b8d7d710..743749210 100644
--- a/src/render/backend/cameralens.cpp
+++ b/src/render/backend/cameralens.cpp
@@ -133,12 +133,12 @@ void CameraLens::syncFromFrontEnd(const Qt3DCore::QNode *frontEnd, bool firstTim
     const Matrix4x4 projectionMatrix(node->projectionMatrix());
     if (projectionMatrix != m_projection) {
         m_projection = projectionMatrix;
-        markDirty(AbstractRenderer::AllDirty);
+        markDirty(AbstractRenderer::ParameterDirty);
     }
 
     if (!qFuzzyCompare(node->exposure(), m_exposure)) {
         m_exposure = node->exposure();
-        markDirty(AbstractRenderer::AllDirty);
+        markDirty(AbstractRenderer::ParameterDirty);
     }
 
     const QCameraLensPrivate *d = static_cast<const QCameraLensPrivate *>(QNodePrivate::get(node));
-- 
2.33.0

